src/lib/Form/Factory/FormFactory.php

```diff
@@ -97,21 +98,19 @@ public function forgotUserPasswordWithLogin(
     }
 
     /**
-     * @param \Ibexa\User\Form\Data\UserPasswordResetData $data
-     * @param string|null $name
-     *
-     * @return \Symfony\Component\Form\FormInterface
-     *
      * @throws \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException
      */
     public function resetUserPassword(
-        UserPasswordResetData $data = null,
+        ?UserPasswordResetData $data = null,
```

Comment:
Can't we enforce $data to be just `UserPasswordData` without `null` if we really don't support `null`?

In the `user` package it's executed only in 1 place where $data is not null

Reply:
You are right, but for the sake of efficiency let's stick to original task which is (getting rid of deprecated syntax instead of evaluating each case separately) :wink: